arangod/Aql/Functions.cpp

```diff
@@ -6507,3 +6518,412 @@ AqlValue Functions::Fail(arangodb::aql::Query* query, transaction::Methods* trx,
+      return std::to_string(seconds);
+    }},
+  {"%f", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto day_time = make_time(tp - floor<days>(tp));
+      uint64_t millis = day_time.subseconds().count();
+      return std::to_string(millis);
+    }},
+  {"%x", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto ymd = year_month_day(floor<days>(tp));
+      auto yyyy = year{ymd.year()};
+      auto firstDayInYear = yyyy / 1 / 0;
+      uint64_t daysSinceFirst = duration_cast<days>(tp - sys_days(firstDayInYear)).count();
+      return std::to_string(daysSinceFirst);
+    }},
+  {"%k", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      iso_week::year_weeknum_weekday yww{floor<days>(tp)};
+      uint64_t isoWeek = static_cast<uint64_t>((unsigned)(yww.weeknum()));
+      return std::to_string(isoWeek);
+    }},
+  {"%l", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      year_month_day ymd{floor<days>(tp)};
+      return (ymd.year().is_leap())? std::string("1"): std::string("0");
+  }},
+  {"%q", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      year_month_day ymd{floor<days>(tp)};
+      month m = ymd.month();
+      uint64_t part = static_cast<uint64_t>(ceil(unsigned(m) / 3.0f));
+      TRI_ASSERT(part <= 4);
+      return std::to_string(part);
+    }},
+  {"%a", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto ymd = year_month_day{floor<days>(tp)};
+      auto lastMonthDay = ymd.year() / ymd.month() / last;
+      return std::to_string(static_cast<uint64_t>(unsigned(lastMonthDay.day())));
+    }},
+  {"%%", [](std::string& wrk, tp_sys_clock_ms const& tp) { return std::string("%"); }},
+  {"%", [](std::string& wrk, tp_sys_clock_ms const& tp) { return ""; }}
+};
+
+std::string regex_replace(std::string const& search,
+                          const std::regex& re,
+                          dateMapType const& dateMap,
+                          tp_sys_clock_ms const& tp)
+{
+  std::string s;
+
+  auto first = search.begin();
+  auto last = search.end();
+  typename std::smatch::difference_type
+    positionOfLastMatch = 0;
+  auto endOfLastMatch = first;
+
+  auto callback = [&tp, &endOfLastMatch, &positionOfLastMatch, &s, &dateMap](const std::smatch& match)
+    {
+      auto positionOfThisMatch = match.position(0);
+      auto diff = positionOfThisMatch - positionOfLastMatch;
+
+      auto startOfThisMatch = endOfLastMatch;
+      std::advance(startOfThisMatch, diff);
+
+      s.append(endOfLastMatch, startOfThisMatch);
+      auto got = dateMap.find(match.str(0));
+      if (got != dateMap.end()) {
+        std::string sanotehu;
+        s.append(got->second(sanotehu, tp));
+      }
+      auto lengthOfMatch = match.length(0);
+
+      positionOfLastMatch = positionOfThisMatch + lengthOfMatch;
+
+      endOfLastMatch = startOfThisMatch;
+      std::advance(endOfLastMatch, lengthOfMatch);
+    };
+
+  std::regex_iterator<std::string::const_iterator> end; /// TODO - howto initialize?
```

Comment:
what about this todo?

Reply:
The default constructed iterator is the end sentinel.
http://en.cppreference.com/w/cpp/regex/regex_iterator/regex_iterator